src/services.rs

```diff
@@ -0,0 +1,93 @@
+use std::sync::{Arc, Mutex};
+use std::collections::HashMap;
+use multimap::MultiMap;
+use rand::{Rng, thread_rng};
+use dns_parser::{self, QueryClass, Name, RRData};
+
+pub type AnswerBuilder = dns_parser::Builder<dns_parser::Answers>;
+
+
+pub type SharedServices = Arc<Mutex<Services>>;
```

Comment:
Also, I'm don't care about performance, but a `RwLock` is probably more appropriate, since this is going to be read a lot more than written